### PR TITLE
fixed enable_avoid_dynamic_calls

### LINKED ISSUE #1721: Enabling `avoid_dynamic_calls` in analysis_options.yaml
**Describe the bug**
As part of PR https://github.com/PalisadoesFoundation/talawa/pull/1510 for upgrading the flutter code upgrade to v3.7.0, we made edits to the [analysis_options.yaml](https://github.com/PalisadoesFoundation/talawa/blob/develop/analysis_options.yaml) file. Although most of these changes have been reverted, we still need to re-enable `avoid_dynamic_calls` and fix the resulting bugs throughout the codebase.

**To Reproduce**
Run the current suite of tests, and they all pass. This is because the `avoid_dynamic_calls` check has been disabled

**Expected behavior**
`avoid_dynamic_calls` must be re-enabled, and all the resulting errors must be fixed.

**Actual behavior**
`avoid_dynamic_calls` is currently disabled.

**Screenshots**
image

## Progress
- [ ] Posts
- [ ] Chats
- [ ] Events
- [ ] Profile
- [ ] Widgets
- [ ] Plugin
- [ ] Comments
- [ ] database mutations
- [ ] Main screen
- [ ] Pre-auth screens
- [ ] Miscellaneous

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Azad99-9**
**What kind of change does this PR introduce?**

This PR fixes the issue enable_avoid_dynamic_calls

**Issue Number:**

Fixes #1721 

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

removed the` ignore directive : avoid_dynamic_call ` from the codebase and fixed the consequential warnings.

**Does this PR introduce a breaking change?**

No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Shouldn't you also include the updated `analysis_options.yaml` file?

**Comment by Azad99-9** [comment]
actually the analysis_options.yaml file is already updated in [this PR](https://github.com/PalisadoesFoundation/talawa/pull/1771).
So now the task of the issue has boiled down to remove the ignore_directives from the entire codebase and fix the respective warnings.

**Comment by palisadoes** [comment]
Thanks. Please work with the assigned reviewers to approve your approach

File: lib/view_model/pre_auth_view_models/login_view_model.dart [hunk 1]
```diff
@@ -60,7 +77,29 @@ class LoginViewModel extends BaseModel {
...
+  /// 6. Initializing database functions.
+  /// 7. Performing a GraphQL mutation to login the user by providing the email and encrypted password.
+  /// 8. Handling the result of the login operation:
+  ///    - Updating the current user with the received data.
+  ///    - Redirecting the user based on their status in the application.
+  ///    - Handling Firebase options for Android and iOS if available.
+  ///    - Configuring Firebase and saving FCM token to the database.
+  ///
+  /// In case of any exceptions during the login process, this function catches and prints the error.
   Future<void> login() async {
```
Comment (literalEval) [inline lib/view_model/pre_auth_view_models/login_view_model.dart:103]: Very good documentation.

File: lib/widgets/pinned_carousel_widget.dart [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
-// ignore_for_file: talawa_good_doc_comments
-
 import 'package:flutter/material.dart';
 import 'package:talawa/models/post/post_model.dart';
 import 'package:talawa/services/size_config.dart';
 import 'package:talawa/utils/app_localization.dart';
 
-/// PinnedPostCarousel class returns a widget for pinned posts in
-/// the slider/carousel on the top of the Home Screen.
+/// PinnedPostCarousel class returns a widget for pinned posts in the slider/carousel on the top of the Home Screen.
```
Comment (literalEval) [inline lib/widgets/pinned_carousel_widget.dart:6]: Shorten this line, and have the explanation in below one.

File: test/view_model_tests/pre_auth_view_models/login_view_model_test.dart [hunk 3]
```diff
@@ -31,123 +39,127 @@ final data = {
...
+        source: QueryResultSource.network,
+        data: data,
+        options: QueryOptions(document: gql(queries.loginUser('', ''))),
+      );
+
+      when(databaseFunctions.gqlNonAuthMutation(queries.loginUser('', '')))
+          .thenAnswer((_) async => result);
+
+      await model.login();
+
```
Comment (literalEval) [inline test/view_model_tests/pre_auth_view_models/login_view_model_test.dart:80]: The newline after await and expect seem extraneous. Please remove them all over the file.

**Review by literalEval — CHANGES_REQUESTED** [review]
Thanks for writing docs properly. Please do these minor changes.

File: lib/view_model/pre_auth_view_models/login_view_model.dart (hunk 1 above)
Comment (Azad99-9) [reply lib/view_model/pre_auth_view_models/login_view_model.dart:103]: Thank you very much for your appreciation. I'm grateful for your recognition of my work.

**Comment by Azad99-9** [comment]
@literalEval @Ayush0Chaudhary @DMills27 kindly review this PR.

File: lib/view_model/pre_auth_view_models/login_view_model.dart (hunk 1 above)
Comment (literalEval) [reply lib/view_model/pre_auth_view_models/login_view_model.dart:103]: Hey @Azad99-9, I updated the comment during my previous review but somehow it didn't. Please check the previous comment again

**Comment by literalEval** [comment]
Please fix the doc as suggested now

**Comment by Azad99-9** [comment]
@literalEval please have a look i have `reformatted` the docs.

**Review by literalEval — CHANGES_REQUESTED** [review]
Please make sure that every doc (and possibly code too), is in the range of 80 characters per line max.

**Comment by literalEval** [comment]
@Azad99-9 please do this quick change.

**Comment by Azad99-9** [comment]
Sure @literalEval

**Comment by Azad99-9** [comment]
@literalEval please have a look, I have updated the comments length.

**Review by literalEval — APPROVED** [review]
LGTM

**Review by palisadoes — APPROVED** [review]

**Pull request merged**